File: src/menus/V2/social.ts
import { createContainer, createLinkButton, createRow, createSeparator, createTextDisplay } from "@magicyan/discord";
import { ContainerBuilder, MessageActionRowComponentBuilder } from "discord.js";

export function socialComponent(): ContainerBuilder {
    return createContainer({
        accentColor: constants.colors.green,
        components: [
            createTextDisplay("**FOLLOW ME ON SOCIAL MEDIA**", 0),
            createSeparator(false, false),
            createRow<MessageActionRowComponentBuilder>().addComponents(
                [
                    createLinkButton({
                        label: "Portifolio",
                        emoji: emojis.static.icons_p,
                        url: "https://nathan-miguel.vercel.app/"
                    }),
                    createLinkButton({
                        label: "Youtube",
                        emoji: emojis.static.icons_youtube,
                        url: "https://www.youtube.com/@NathanMiguel1"
                    }),
                    createLinkButton({
                        label: "Instagram",
                        emoji: emojis.static.icons_instagram,
                        url: "https://www.instagram.com/nathan_cmiguel/"
                    }),
                    createLinkButton({
                        label: "Github",
                        emoji: emojis.static.icons_github,
                        url: "https://github.com/Cesio137"
                    }),
                    createLinkButton({
                        label: "X/Twitter",
                        emoji: emojis.static.icons_x,
                        url: "https://x.com/NathanCmig"
                    }),
                ]
            ),
            createSeparator(true, true),
            createTextDisplay("**VISIT MY FAB STORE**", 1),
            createSeparator(false, false),
            createLinkButton({
                label: "Fab",
                emoji: emojis.static.icons_f,
                url: "https://www.fab.com/sellers/Nathan%20Miguel"
            }),
        ]
    })
}

File: src/menus/components/social.rs
use crate::data::*;
use serenity::all::{
    Colour, CreateActionRow, CreateButton, CreateComponent, CreateContainer, CreateSeparator,
    CreateTextDisplay, EmojiId, ReactionType, Spacing,
};
use std::borrow::Cow;

pub fn social_component<'a>() -> CreateComponent<'a> {
    let accent_color = Colour::new(str_hex_to_u32(&CONSTANTS.colors.green));
    let text_display = CreateTextDisplay::new("**FOLLOW ME ON SOCIAL MEDIA**");
    let separator = CreateSeparator::new(false).spacing(Spacing::Small);

    let social_row = CreateActionRow::Buttons(Cow::Owned(vec![
        CreateButton::new_link("https://nathan-miguel.vercel.app/")
            .label("Portifolio")
            .emoji(ReactionType::Custom {
                id: EmojiId::from(str_to_u64(&EMOJIS.emojis_static.icons_p)),
                name: None,
                animated: false,
            }),
        CreateButton::new_link("https://www.youtube.com/@NathanMiguel1")
            .label("Youtube")
            .emoji(ReactionType::Custom {
                id: EmojiId::from(str_to_u64(&EMOJIS.emojis_static.icons_youtube)),
                name: None,
                animated: false,
            }),
        CreateButton::new_link("https://www.instagram.com/nathan_cmiguel/")
            .label("Instagram")
            .emoji(ReactionType::Custom {
                id: EmojiId::from(str_to_u64(&EMOJIS.emojis_static.icons_instagram)),
                name: None,
                animated: false,
            }),
        CreateButton::new_link("https://github.com/Cesio137")
            .label("GitHub")
            .emoji(ReactionType::Custom {
                id: EmojiId::from(str_to_u64(&EMOJIS.emojis_static.icons_github)),
                name: None,
                animated: false,
            }),
        CreateButton::new_link("https://x.com/NathanCmig")
            .label("X/Twitter")
            .emoji(ReactionType::Custom {
                id: EmojiId::from(str_to_u64(&EMOJIS.emojis_static.icons_x)),
                name: None,
                animated: false,
            }),
    ]));

    let fab_separator = CreateSeparator::new(true).spacing(Spacing::Large);
    let fab_text_display = CreateTextDisplay::new("**VISIT MY FAB STORE**");

    let fab_row = CreateActionRow::Buttons(Cow::Owned(vec![
        CreateButton::new_link("https://www.fab.com/sellers/Nathan%20Miguel")
            .label("Fab")
            .emoji(ReactionType::Custom {
                id: EmojiId::from(str_to_u64(&EMOJIS.emojis_static.icons_f)),
                name: None,
                animated: false,
            }),
    ]));

    CreateComponent::Container(
        CreateContainer::new(vec![
            CreateComponent::TextDisplay(text_display),
            CreateComponent::Separator(separator.clone()),
            CreateComponent::ActionRow(social_row),
            CreateComponent::Separator(fab_separator.clone()),
            CreateComponent::TextDisplay(fab_text_display),
            CreateComponent::Separator(fab_separator.divider(false).spacing(Spacing::Small)),
            CreateComponent::ActionRow(fab_row),
        ])
        .accent_color(accent_color),
    )
}
